torcheeg/trainers/classifier.py

```diff
@@ -266,4 +296,349 @@ def predict_step(self,
+            model = MyModel()
+        
+        Use :obj:`nn.ModuleDict` is a simplest way to define model.
+
+        .. code-block:: python
+
+            model = nn.ModuleDict( {'decoder': nn.Linear(10,5),'predict_by_feature': nn.Linear(5,2)})
+
+                    
+        And it is also feasible to define :obj:`decorder` and :obj:`predict_by_feature` method in your model.  
+
+        .. code-block:: python
+
+            class MyModel(nn.Module):
+                def __init__(self) -> None:
+                    super(md,self).__init__()
+                    self.layer1 = nn.Linear(10,5)
+                    self.layer2 = nn.Linear(5,2)
+                
+                def decoder(self,x):
+                    return self.layer1(x)
+                
+                def predict_by_feature(feature):
+                    return self.layer2(feature)
+    
+                def forward(self,x):
+                    return self.predict_by_feature(self.decoder(x))
+            model = MyModel()
+        
+
+        Then we simply pass the model to init trainer. Done!
+        
+        .. code-block:: python
+
+            trainer = ClossClassifierTrainer(model)
+            trainer.fit(train_loader, val_loader)
+            trainer.test(test_loader)
+
+        Args:
+            model (nn.Module): model (nn.Module): The classification model, which should provide the :obj:`".decoder()"` , :obj:`".predict_by_feature()"` method or the model  should contain decoder block(nn.Module), predict block(nn.Module) attributes and ensure this two attirbutes are named "decoder","predict_by_feature".
+            center_dim (int): The dimemsion of decoder output code whose mean values we can loosely regard as the "center". If you don't specify this value, trainer will automatically infer this value when we first pass dataloader into :obj:`trainer.fit()` or :obj:`trainer.test()` method. (defualt: :obj:`None`) 
+            num_classes (int, optional): The number of categories in the dataset. If :obj:`None`, the number of categories will be inferred from the attribute :obj:`num_classes` of the model. (defualt: :obj:`None`)
+            lammda (float): The weight of the center loss in total loss. (default: :obj:`5e-4`)
+            lr (float): The learning rate. (default: :obj:`0.001`)
+            weight_decay (float): The weight decay. (default: :obj:`0.0`)
+            devices (int): The number of devices to use. (default: :obj:`1`)
+            accelerator (str): The accelerator to use. Available options are: 'cpu', 'gpu'. (default: :obj:`"cpu"`)
+            metrics (list of str): The metrics to use. Available options are: 'precision', 'recall', 'f1score', 'accuracy'. (default: :obj:`['accuracy', 'precision', 'recall', 'f1score']`)
+
+        .. automethod:: fit
+        .. automethod:: test
+    '''
+    def __init__(self,
+                 model: nn.Module,
+                 center_dim: int = None,
```

Comment:
Confusing, `hid_channels`

Reply:
This is not hid_channel . Is it appropriate to call it hid_dim?

Reply:
I didn't get it; could you please explain why 'This is not hid_channel'? Isn't it the hidden dimension (named by channels as torch layers) of the features extracted by the extractor?